File: NUMBER_and_KSTRPOW/contestants/huynh_minh_thuan/KSTRPOW.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ifstream inp("KSTRPOW.inp");
    ofstream out("KSTRPOW.out");

    string s, t;
    int k;

    inp >> s >> t >> k;

    string str = s;

    for (int i = 1; i < k; i++) str += s;

    if (str == t) out << "YES";
    else out << "NO";

    return 0;
}

File: NUMBER_and_KSTRPOW/contestants/nguyen_the_khoi/kstrpow.py
with open ("kstrpow.inp","r") as f:
    s=f.readline().strip()
    t=f.readline().strip()
    k=int(f.readline())
    f.close()
a=len(s)
b=len(t)
with open ("kstrpow.out","w") as f:
    if a*k==b:
        f.write("YES")
    else:
        f.write("NO")
    f.close()